### ISSUE TITLE
DropDownButton/SplitButton throw an ExpressionChange error when opening Dialog using routing

### ISSUE THREAD

**Issue by mbechev** [telerik/kendo-angular#3778]
**Describe the bug**
DropDownButton and SplitButton components throw _ExpressionChange_ error when opening a Dialog which is defined in a parent component that has defined `<router-outlet>`. 

![image](https://user-images.githubusercontent.com/33104481/191239749-5a588bb8-03b0-481b-b43b-a8a7088f3543.png)

_app.component.ts_
The order of the `<router-outlet>` and `kendoDialogContainer` seems important, as swapping them fixes the issue.

```ts
@Component({
  selector: "app-root",
  template: `
          <router-outlet></router-outlet>
	  <div kendoDialogContainer></div>
	`,
})
export class AppComponent {}
```

_nested.component.ts_
```ts
@Component({
  selector: "app-nested",
  template: `
    <kendo-dropdownbutton [data]="data"> User Settings </kendo-dropdownbutton>
  `,
})
export class NestedComponent {
  public data = [
    {
      text: "My Profile",
      click: () => this.open(),
    },
  ];
  constructor(private dialogService: DialogService) {}
  open(): void {
    this.dialogService.open({
      title: "My Dialog",
      content: `Dialog Content`,
    });
  }
}
```

**To Reproduce**
[dropdownbtn-issue.zip](https://github.com/telerik/kendo-angular/files/9606825/dropdownbtn-issue.zip)

1. Download and unzip the file.
2. Run `npm install`.
3. Execute `ng serve`.
4. Click the DropDownButton and select the first option to open the Dialog.

Defining the `kendoDialogContainer` in the _nested-component.ts_ file also fixes the issue.

**Comment by michaelmarcuccio** [comment]
It also occurs in other scenarios with inline <kendo-dialog> being toggled. I am having a very hard time reproducing this specifically, but it occurs in an app I cannot share. Just letting you know it is not isolated to just the dialogService/router-outlet causing trouble.

**Comment by michaelmarcuccio** [comment]
I created a workaround for this by blurring the dropdownbutton on click:
```
import { Directive } from '@angular/core';
import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
import { DropDownButtonComponent } from '@progress/kendo-angular-buttons';

@UntilDestroy()
@Directive({ selector: '[ddbFix]' })
export class KendoDDBUnfocusFixDirective {
  constructor(host: DropDownButtonComponent) {
    host.itemClick.pipe(untilDestroyed(this)).subscribe(() => {
      host.blur();
    });
  }
}
```

**Comment by Raisolution** [comment]
Fixed in the latest develop version `v16.2.0-develop.1`
